go/vt/sqlparser/comments.go

```diff
@@ -258,8 +260,16 @@ func (d CommentDirectives) IsSet(key string) bool {
 	if d == nil {
 		return false
 	}
-	val, ok := d[key]
-	if !ok {
+	var val string
+	exists := false
+	for commentDirective, value := range d {
+		if strings.EqualFold(commentDirective, key) {
+			val = value
+			exists = true
+			break
+		}
+	}
+	if !exists {
```

Comment:
I think we can avoid this check by setting the map key in one defined case like `lower_case` and also checking the key in `lower_case`.

Reply:
👍